tools/querytee/proxy.go

```diff
@@ -305,7 +307,10 @@ func (p *Proxy) Start() error {
 		if p.cfg.CompareResponses {
 			comparator = route.ResponseComparator
 		}
-		router.Path(route.Path).Methods(route.Methods...).Handler(NewProxyEndpoint(p.backends, route, p.metrics, p.logger, comparator, p.cfg.LogSlowQueryResponseThreshold, p.cfg.SecondaryBackendsRequestProportion))
+		router.Path(route.Path).Methods(route.Methods...).Handler(
+			NewProxyEndpoint(p.backends, route, p.metrics, p.logger, comparator, p.cfg.LogSlowQueryResponseThreshold, p.cfg.SecondaryBackendsRequestProportion).
+				WithSkipPreferredBackendFailures(p.cfg.SkipPreferredBackendFailures),
```

Comment:
Given the pattern so far has been to pass config as an argument to `NewProxyEndpoint`, is there a reason to introduce the `With...` pattern here?

Reply:
It was getting long 🤷. I don't mind either way

Reply:
I think I'd prefer to either keep it consistent or change it for all of the options, with a slight preference for keeping things consistent with how they were to reduce the size of this PR

Reply:
👍 Done!